features/FxA-107-security-event-history/README.md

```diff
@@ -6,29 +6,200 @@ As a Firefox Accounts user, I want to reduce the amount of sign-in
+as a reliable heuristic for detecting legitimate login requests,
+and to use that to avoid sending a sign-in confirmation email.
+
+We will initial focus on gathering metrics
+on the reliability of this heuristic,
+and will only proceed with skipping the confirmation email
+if the metrics prove promising.
+
+## Hypothesis
+
+We believe that skipping the sign-in confirmation email
+for users who are logging in from an IP address they have previously used,
+will remove friction from the login process
+for legitimate users,
+without increasing the potential for fraudulent logins.
+
+We will know this is true when we see
+an increase in the sign-in success rate
+and a reduction in the median time to successful login,
+that are not accompanied by a noticeable increase
+in the rate at which fradulent logins are reported.
 
 ## Metrics
 
-We can log an info message each time this would occur, tagged as
-`Account.history.seen`. A graph can be created, comparing:
+### Key Success Metrics
+
+The key success metrics for this feature
+are the following existing KPIs:
+
+* sign-in success rate
+* median time to successful sign-in
+* rate of logins reported as fraudulent
+
+These will tell us whether the feature
+is impacting the things we care about.
+
+### Secondary Success Metrics
+
+The operation of this feature will be monitored
+in more detail by emitting the following metrics:
+
+* `Account.history.verified`
+  * Emitted when a successful login is made from an IP
+    address used for a previous, verified login by that user.
+    Includes a `recency` field with the following values:
+    * `day` - the IP was used less than 1 day ago
+    * `week` - the IP was used less than 1 week ago
+    * `month` - the IP was used less than 1 month ago
+    * `old` - the IP was used more than 1 month ago
+* `Account.history.unverified`
+  * Emitted when a successful login is made from an IP
+    address used for a previous, unverified login by that user.
+* `Account.history.false_positive`
+  * Emitted when a login matched `Account.history.verified`,
+    but was later reported to have been fraudulent.
+    Includes a `recency` field in the same manner as
+    the `Account.history.verified` event.
+* `Account.history.error`
+  * Emitted whenever there is an error processing the
+    IP history
+
+We'll graph the following to judge the ongoing performance
+of this feature:
+
+* Hit rate: `Account.history.verified`
+  as a percentage of all successful logins.
+  This verifies whether the feature is providing value
+  to a significant number of our users.
+* False-positive rate: `Account.history.false_positive`
+  as a percentage of `Account.history.verified`.
+  This verifies whether the feature is indeed
+  having a minimal impact on the potential for fraudulent logins.
+
+Each graph will be split by
+the different values of `recency`
+on the events, like this:
+
+<img src="hit_rate.jpg" />
+
+<img src="false_positive_rate.jpg" />
```

Comment:
Luckily for me, my kids have a big box of coloured pens.  Unluckily for me, only about a quarter of them actually seem to work.  So this was also a great excuse to throw out a bunch of broken pens!